internal/characters/xiao/skill.go

```diff
@@ -77,3 +79,17 @@ func (c *char) Skill(p map[string]int) action.ActionInfo {
 		State:           action.SkillState,
 	}
 }
+
+func (c *char) makeParticleCB() combat.AttackCBFunc {
+	done := false
+	return func(a combat.AttackCB) {
+		if a.Target.Type() != combat.TargettableEnemy {
+			return
+		}
+		if done {
+			return
+		}
+		done = true
+		c.Core.QueueParticle(c.Base.Key.String(), 3, attributes.Anemo, c.ParticleDelay)
```

Comment:
from dm: 0.5s cooldown rather than once per dash. Preview gif on fandom appears to demonstrate this (even though 0.5s cooldown not mentioned in their description)

https://static.wikia.nocookie.net/gensin-impact/images/9/9f/Lemniscatic_Wind_Cycling_Ground_Preview.gif

Reply:
If it was a 0.5s cd applied to xiao then he wouldn't be able to generate two sets of particles from doing 2Es in a row:
0: 1st E start
4: 1st E hitmark
24: 2nd E start
28: 2nd E hitmark

28-4 = 24f < 30f particle icd